src/main/java/frc/robot/RobotContainer.java

```diff
@@ -548,6 +563,36 @@ private void burnFlashAllSparks() {
     Timer.delay(0.25);
   }
 
+  /**
+   * @return true when the currently prepped state is ready to score
+   */
+  public boolean readyToScoreCheck() {
+    switch (prepState) {
+      case OFF:
+        return false;
+      case CLIMB:
+        return intake.nearDeployed() && elevator.atDesiredPosition() && shooter.atDesiredPosition();
+      case SPEAKER:
+        return elevator.atDesiredPosition()
+            && shooter.atDesiredPosition()
+            && shooter.isShooterSpunUp()
+            && drive.getDiffCurrentTargetYawDeg()
+                < ShooterCal.ROBOT_HEADING_MARGIN_TO_SHOOT_DEGREES;
```

Comment:
Currently this should check if we've set a heading based on a tag

Reply:
Want to make sure this is correct before I implement it: I don't think we currently have a "using heading based on tag" check anywhere? So we would need a public boolean in drivesubsystem, set that to true in speaker prep (if it's actually using a tag), then set it to false... if the heading or position changes because of the driver (maybe in `drive()`)? Then just check that here.

Reply:
Should it also continue to check if yaw error is less than the margin?

Reply:
> So we would need a public boolean in drivesubsystem, set that to true in speaker prep (if it's actually using a tag), then set it to false... if the heading or position changes because of the driver (maybe in drive())? Then just check that here.

Yeah something like that. Maybe set before and after SpeakerPrep in the associated button's sequence?

> Should it also continue to check if yaw error is less than the margin?
Yeah we need to have seen the tag and we need to be aligned to it

Reply:
Done